src/main/java/org/gradle/profiler/asyncprofiler/AsyncProfilerWallProfilerFactory.java

```diff
@@ -0,0 +1,38 @@
+package org.gradle.profiler.asyncprofiler;
+
+import com.google.common.collect.ImmutableList;
+import joptsimple.OptionSet;
+import org.gradle.profiler.OperatingSystem;
+import org.gradle.profiler.Profiler;
+
+import java.util.List;
+
+public class AsyncProfilerWallProfilerFactory extends AsyncProfilerFactory {
+    @Override
+    public Profiler createFromOptions(OptionSet parsedOptions) {
+        AsyncProfilerConfig config = super.createConfig(parsedOptions);
+
+        // Combined cpu + wall events are not supported on macOS at this time
+        List<String> allEvents = OperatingSystem.isMacOS()
+            ? ImmutableList.of("wall")
+            : ImmutableList.of("cpu", "wall");
+        AsyncProfilerConfig overrides = new AsyncProfilerConfig(
+            config.getDistribution(),
+            allEvents,
+            AsyncProfilerConfig.Counter.SAMPLES,
+            config.getInterval(),
+            config.getAllocSampleSize(),
+            config.getLockThreshold(),
+            config.getWallInterval(),
+            config.getStackDepth(),
+            config.isIncludeSystemThreads(),
+            AsyncProfilerOutputType.JFR
```

Comment:
How come `AsyncProfilerAllEventsProfilerFactory` uses `getPreferredOutputType` instead of JFR, as it also has multiple event types? I feel like both `all` and `wall` should consistently be using the same parameter here.